### ISSUE TITLE
Color scheme improvements request for records and functions (OCaml)

### ISSUE THREAD

**Issue by PaulRaUnite** [giraud/reasonml-idea-plugin#455]
plugin version: 0.121-2023.3

### Description
First of all, thank you for the plugin, it seems to be the only one supporting multi-file usage search.

The problem I have is that the code looks really bland and it is hard to read it. More precisely, the color scheme doesn't allow to distinguish function calls, declaration of record fields and their access. I attach some code as an example (code taken from https://ocaml.org/p/camllib/). 
![image](https://github.com/giraud/reasonml-idea-plugin/assets/16649084/73b4b220-984b-41aa-a9d3-6acf3f6797de)
It seems that the plugin already knows about the record field and what is a function, what could probably be a problem is to understand when the function is called and when it is passed as a parameter.

Another minor issue is that it seems that the plugin does not recognize optional and labelled parameters. At least they are not highlighted as it is with other parameters (article https://ocaml.org/docs/labels has some quick examples if you would want to test it).

**Comment by giraud** [comment]
Hi, thank you for the appreciation.

It looks like you are using the default theme, but it is already possible to make the code less bland.
You can go to `Settings` | `Editor` | `Color scheme` | `Rescript/Reason (OCaml)` and update colors to your liking.

For example (colors are bad, just to highlight everything):
![image](https://github.com/giraud/reasonml-idea-plugin/assets/239612/d14d028e-9a0f-4fda-bf30-bdf0d9c8faa2)

**Comment by giraud** [comment]
I think I can easily add highlighting for a record field in the definition.

what you ask looks like similar to https://github.com/giraud/reasonml-idea-plugin/discussions/147.
I could add a color highlighting for the name of a let/var (as seen here https://ocaml.org/docs/labels)

But it can be difficult to know what the lower symbol is (function, field, type) without too much performance penalty.

for parameter labels, do you expect that the pos and len can be highlighted differently ?
![image](https://github.com/giraud/reasonml-idea-plugin/assets/239612/332442ae-4e12-4741-bcdf-d9f14a95efdc)

**Comment by PaulRaUnite** [comment]
> what you ask looks like similar to #147.

Sorry, I completely missed the discussions feature, I should've written there.

> I think I can easily add highlighting for a record field in the definition. I could add a color highlighting for the name of a let/var (as seen here https://ocaml.org/docs/labels)

Yeah, it would be nice.

> But it can be difficult to know what the lower symbol is (function, field, type) without too much performance penalty.

You are right, now thinking about it, something like the Rust plugin spends some time indexing at the startup, which seems to be not the case for ReasonML plugin.

I guess rich color scheme would require something like this to be implemented, and if you don't have any plans for on-the-fly lints or type inference, it is hard to justify doing so much work for some contrast, I understand.

> for parameter labels, do you expect that the pos and len can be highlighted differently? It can be done easily at the declaration, but inside the let body it can be trickier

Not really. What I meant is that when you place the cursor on the regular argument, its usage in the function body is highlighted, and for labelled arguments it doesn't always happen. At least for the first argument, it is the label that gets highlighted.

**Comment by giraud** [comment]
> Not really. What I meant is that when you place the cursor on the regular argument, its usage in the function body is highlighted, and for labelled arguments it doesn't always happen. At least for the first argument, it is the label that gets highlighted.

ok, I got confused because this is not color scheme, but symbol resolution.
It is a bug, parameters references are not correctly resolved to their declaration. If not highlithed correctly it means the references cannot be resolved to the declaration and many IDE features wont work like navigation, completion, renaming, etc.

I see that if a function declaration is using a labelled parameter first, parsing is incorrect (I created #456)
